src/commands/recover.py

```diff
@@ -0,0 +1,231 @@
+from src.common.password import generate_password, get_or_create_password_file
+from src.common.utils import log_verbose
+from src.common.validators import validate_eth_address, validate_mnemonic
+from src.common.vault_config import VaultConfig
+from src.config.settings import AVAILABLE_NETWORKS, GOERLI, settings
+
+
+@dataclass
+class RegisteredValidator:
+    index: int
+    public_key: HexStr
+    status: str
+    keystore_found: bool = field(default=False)
+
+
+@click.command(help='Recover vault data directory and keystores.')
+@click.option(
+    '--data-dir',
+    default=str(Path.home() / '.stakewise'),
+    envvar='DATA_DIR',
+    help='Path where the vault data will be placed. Default is ~/.stakewise.',
+    type=click.Path(exists=False, file_okay=False, dir_okay=True),
+)
+@click.option(
+    '--per-keystore-password',
+    is_flag=True,
+    default=False,
+    help='Creates separate password file for each keystore.',
+)
+@click.option(
+    '--mnemonic',
+    help='The mnemonic for generating the validator keys.',
+    prompt='Enter the mnemonic for generating the validator keys',
+    type=str,
+    callback=validate_mnemonic,
+)
+@click.option(
+    '--vault',
+    prompt='Enter your vault address',
+    help='Vault address',
+    type=str,
+    callback=validate_eth_address,
+)
+@click.option(
+    '--execution-endpoints',
+    type=str,
+    envvar='EXECUTION_ENDPOINTS',
+    prompt='Enter comma separated list of API endpoints for execution nodes',
+    help='Comma separated list of API endpoints for execution nodes.',
+)
+@click.option(
+    '--consensus-endpoints',
+    type=str,
+    envvar='CONSENSUS_ENDPOINTS',
+    prompt='Enter comma separated list of API endpoints for consensus nodes',
+    help='Comma separated list of API endpoints for consensus nodes.',
+)
+@click.option(
+    '--network',
+    default=GOERLI,
+    help='The network of your vault.',
+    prompt='Enter the network name',
+    type=click.Choice(
+        AVAILABLE_NETWORKS,
+        case_sensitive=False,
+    ),
+)
+# pylint: disable-next=too-many-arguments
+def recover(
+    data_dir: str,
+    vault: HexAddress,
+    network: str,
+    mnemonic: str,
+    consensus_endpoints: str,
+    execution_endpoints: str,
+    per_keystore_password: bool,
+) -> None:
+    # pylint: disable=duplicate-code
+    config = VaultConfig(
+        vault=vault,
+        data_dir=Path(data_dir),
+    )
+    if config.vault_dir.exists():
+        raise click.ClickException(f'Vault directory {config.vault_dir} already exists.')
+
+    keystores_dir = config.vault_dir / 'keystores'
+    password_file = keystores_dir / 'password.txt'
+
+    settings.set(
+        execution_endpoints=execution_endpoints,
+        consensus_endpoints=consensus_endpoints,
+        vault=vault,
+        network=network,
+        vault_dir=Path(data_dir, vault),
+    )
+
+    try:
+        asyncio.run(
+            main(
+                mnemonic,
+                str(keystores_dir),
```

Comment:
we can use path by default and cast to str only in function like `.generate_credential(`